src/sentry/search/events/filter.py

```diff
@@ -851,6 +852,50 @@ def _environment_filter_converter(
         else:
             return env_conditions[0]
 
+    def _project_slug_filter_converter(
+        self,
+        search_filter: SearchFilter,
+        _: str,
+    ) -> Optional[WhereType]:
+        """Convert project slugs to ids and create a filter based on those
+
+        This is cause we only store project ids in clickhouse
+        """
+        value = search_filter.value.value
+
+        if search_filter.operator == "=" and value == "":
+            raise InvalidSearchQuery(
+                'Cannot query for has:project or project:"" as every event will have a project'
```

Comment:
reworded from the original message of `Invalid query for 'has' search: 'project' cannot be empty.`